### DISCUSSION TITLE
Some sugestions

### DISCUSSION THREAD

**Discussion by ghost** [quer/the-steam-awards#17] [Ideas]
Hello i'm here i have some ideas to new modules,

A module that gives an steam award,
A module to craft badge (example winter sale badge),
A module to create booster packs,
The module "chanceAccountSettings.js" don't work here, i think is because new profile update (i tried change nickname of my accounts but dont work, i receive on console "no chance")

I believe they are good modules, what do you think?

**Comment by quer** [comment]
hey thank for the feedback.

1. The award is a good idea. but after the new winter sale update. There is many places you can give award. it might be a big project. also to know what award to give, if some account do not have the points. it will be hard to explain, people how it works. but it not a bad idea.
2. Most of what i know, use the script to get card, and then transfer them to a other account. but i will create a module to craft all badge that is done. and one to craft a individual badge
3. Im not sure what you are thinking on. do you mean a module to open all booster packs?
4. You are right, I need to update the "chanceAccountSettings" module. it was made to work whit the old steam design. 

I will fix the "chanceAccountSettings" modules, but it will must lightly first be fixed in the new year. I like the award idea. but im not sure how i whoud show the people that use the module how to set it up.

**Comment by ghost** [comment]
Hello,

1 - I believe that award through profile and screenshots/artworks similar to "LikeImageList.js" would be the best way to distribute and can create a list of awards example:
fileid: "123456789"
mad_scientist: 1
extra_helpful: 1
golden_unicorn: 0
1 for give and 0 for dont give

2 - great idea, really, most people transfer the cards to the main account, but there are people who buy  card sets to level up some alternative accounts

3 - I was referring to the creation of booster packs through this link: https://steamcommunity.com//tradingcards/boostercreator/ . For example, there are people who have gems distributed in alternative accounts, because you can only create one booster pack per day for a specific game in each account, for example I can only create one booster pack for csgo per day

Thanks for your attention and Merry Christmas!

**Reply by quer** [reply]
hey, 

1. that might be true, I will look into it, and create it, it if just like that. 
2. true, I will look into it. :)
3. that it right, I have forgotten that exists. I might create that. 

One of my account can give award, as they do not have point, and do not have full set of card. to create and test to see if it works, so i will first look into it when I have the opportunity, to also test it.

In the wiki I have created a tutorial to create your own module if you need somefing that do not exist, then share the module with all. :)
[Wiki New Module Info](https://github.com/quer/the-steam-awards/wiki/New-Module-Info)

Merry Christmas

**Comment by spookyuser** [comment]
Just want to say this is awesome! Would be cool if it could run as a github action 😎

**Reply by quer** [reply]
Thank. 
Im not sure how that can be done. as you will have to upload you account to github, where all can see them. 

but i like the idea, to just use the github actions, to queue. the modules to run. 
you can maby create a fork and set that private. im not sure if that is a thing.

**Reply by spookyuser** [reply]
Yeah you can definitely do that, or we could maybe just set the credentials to read from environmental variables, and then users could set them as secrets in their forked repository settings!

**Reply by quer** [reply]
nice I will look into it, some day. I have added it to my "Nice to have" list. but as I also work on [Steam bot controller](https://github.com/quer/Steam-bot-Controller), that is made to run on a server all the time. and that script, will in time be able to run the modules from this, in runtime.

**Reply by spookyuser** [reply]
Wow that looks awesome, thanks!

**Comment by quer** [comment]
Hey,
just a small update, i have fixed the module to chance the account setting. i have just remade it into some new modules, to work whit the new design. 

I have also made a module to give award.

**Reply by stickerx** [reply]
@quer  Hi

I want to ask you about API key should setup for every account or what 

because I see errors when trying running  steamAwardNominateGame_v2 or gameRecommend

**Reply by quer** [reply]
hey, im not sure what your asking. 
But it sound like this:
https://github.com/quer/the-steam-awards/issues/15

**Reply by stickerx** [reply]
@quer thank you
For example, I have 2 accounts, where should I put 2 Api in steamAwardNominateGame_v2 and gameRecommend

These two scripts above do not work for me help please

**Reply by quer** [reply]
unfortunately the way it works, is it expect the account to be public, in the games. So it made to only take one ApiKey. and then use that key, to fetch all accounts settings. 

A way you can do it. is to add the apiKey to the config file like:
```js
config.push({
	steam_user: "**",
	steam_pass: "**",
	sharedSecret: "**",
        apiKey: "**"
});
```

and in the main.js, add it to the options, ( it the object that is passed into all modules. 
```js
var options = {
					Index: index,
					UserName: auth.steam_user,
					steamUser: steamUser,
					steamFriends: steamFriends,
					accountPretty: steamClient.steamID + " - " + auth.steam_user + ":",
 					ApiKey: auth.apiKey
				}
```

then you can in the module access the apikey for each user. Just replace:
```js
if(apiKey == "xxxxx"){
		console.log("you need to setup your api key. or remove this and skip from the GetMostPlayedGame methode")
		callback();
		return;
	}
```
whit : 
```js
apiKey  = options.ApiKey
```

Ps, i did not test this. but it shoud work just fine

**Reply by stickerx** [reply]
I use APIKEY in one account And then follow your step 

show that error

**Reply by quer** [reply]
remove the "->" that was just for showing what i added. as the rest was as it is.   ( also add the "," at the end of the line above it )

**Reply by stickerx** [reply]
now I see this error

**Reply by stickerx** [reply]
can you upload me the correct file  steamAwardNominateGame_v2.js & main.js

**Comment by stickerx** [comment]
@quer  quick question, please 

AddGameToWishlist  work only with the first account 

and show me done whit a steam user 

I need him to do that all accounts

**Reply by quer** [reply]
what apps are you trying to use?

**Reply by quer** [reply]
From what i can tell, this shoud just work. and as it work for the first account, it uses the same logic for each account, so as it works for one if them, it shoud work for all. 

after: 
`}, function (postErr, postHttpResponse, postBody) {`
add:
`console.log("Error: " + postBody);` 
that is before:
`console.log("added: " + list[index]);`

then run it agirn. then it will also print the response from steam.

**Reply by quer** [reply]
cant tell why this shoud not work, You are sure it did work for the first account, then when it did the second, it did not work anymore ? 

then i might have to look into why i do not work, i do not have the time, to test  this my self. so cant tell if this fail for me also.

**Reply by stickerx** [reply]
befor i added `console.log("Error: " + postBody);
`
was work for the first account only

**Reply by stickerx** [reply]
the probleme was from `modules.push(require('./modules/joinGroup.js'));`

after remove it from main.js

work

**Reply by stickerx** [reply]
@quer 
`
modules.push(require('./modules/joinGroup.js'));`


```
Connected to Steam.
websession start
undefined:1
<html><head><title>Forbidden</title></head><body><h1>Forbidden</h1>Access is denied. Retrying will not help. Please verify your <pre>key=</pre> parameter.</body></html>
^

SyntaxError: Unexpected token < in JSON at position 0
    at JSON.parse (<anonymous>)
    at Request._callback (C:\Users\Meneow\Desktop\node main\modules\joinGroup.js:101:25)
    at Request.self.callback (C:\Users\Meneow\Desktop\node main\node_modules\request\request.js:186:22)
    at Request.emit (events.js:315:20)
    at Request.<anonymous> (C:\Users\Meneow\Desktop\node main\node_modules\request\request.js:1163:10)
    at Request.emit (events.js:315:20)
    at IncomingMessage.<anonymous> (C:\Users\Meneow\Desktop\node main\node_modules\request\request.js:1085:12)
    at Object.onceWrapper (events.js:421:28)
    at IncomingMessage.emit (events.js:327:22)
    at endReadableNT (_stream_readable.js:1220:12)
```

**Reply by quer** [reply]
i do not provide this muth support. as this hole system, is made to run a script on all account. and a easy way to make them. 
You can read the wiki, on how it works. how to easy make what you need. like the module you are using.
Then i just made a some module, to do some thing. And the module you now using, ekstemt the account to be public, and that you entered at apikey. 

that module do more then just join a group, i do 3 think
 * join the once in the list "MustHave"
 * join the level group, that the account can. ( where the api key is used. )
 * random join a group in the "list" 

so if you just whant to join a group, all you need is 
```js
var list = [
    "https://steamcommunity.com/groups/NewSteamCommunityBeta",
    "https://steamcommunity.com/groups/Infinity-CS-GO",
    "https://steamcommunity.com/groups/bigpicture"
    ANd so on..ANd.
]
var RequestCommunity;
var SessionID;
module.exports = async function(steamClient, _RequestCommunity, RequestStore, _SessionID, options, callback){
    RequestCommunity = _RequestCommunity;
    SessionID = _SessionID;
    for (let i = 0; i < list.length; i++) {
        const groupUrl = list[i];
        await JoinGroup(groupUrl)
    }
    callback()
}

function JoinGroup(groupUrl) {
    return new Promise(function (resolve) {
        RequestCommunity.post({
            url: groupUrl,
            form:{
                sessionID: SessionID,
                action: "join"
            }
        }, function (error, response, body) {
            setTimeout(function () {
                resolve();
            }, 500);
        });
    })
}
```
i did not test the code above. but that is the ideer. how to each make all account join the list of groups.
